packages/hardhat-zksync-vyper/src/index.ts

```diff
@@ -116,3 +125,37 @@ subtask(TASK_COMPILE_VYPER_GET_BUILD, async (args: { vyperVersion: string }, hre
 
     return vyperBuild;
 });
+
+// This task is overriden since TASK_COMPILE_VYPER_LOG_COMPILATION_RESULT logs both solidity and vyper compiled contracts:
+subtask(TASK_COMPILE_SOLIDITY_LOG_COMPILATION_RESULT, async () => {});
+
+subtask(TASK_COMPILE_VYPER_LOG_COMPILATION_RESULT, async ({ versionGroups, quiet }, hre) => {
+    const vyperCompilationsNum = Object.entries(versionGroups).length;
+
+    if (quiet || vyperCompilationsNum === 0) return;
+
+    let vyperNum = 0;
+    let solcNum = 0;
+
+    // We can ask for all compiled contracts since this task is run after all other compilation tasks for both solc and vyper
+    const allContractName = await hre.artifacts.getAllFullyQualifiedNames();
+
+    allContractName.forEach((fullyQualifiedName) => {
+        const contractName = fullyQualifiedName.split(':')[0];
+        if (contractName.slice(-3) === VYPER_EXTENSION) vyperNum++;
+        else if (contractName.slice(-4) === SOLC_EXTENSION) solcNum++;
+    });
+
+    if (solcNum != 0 && vyperNum != 0) {
```

Comment:
Shouldn't there be something similar on the zksolc side?

Reply:
We are already addressing the zksolc logging here, since we want to output both so solc and vyper compilation at the same time. If there is no output for zksolc compilation, it will report "Nothing to compile" as expected.

Reply:
But what happens when there are only solidity files to compile?

Reply:
If there are only the solidity files there it doesn't make much sense to import/use vyper plugin.
I will add a warning message to user about this.